rfc/image-discoverability.md

```diff
@@ -0,0 +1,127 @@
+    gcpmeasurements --> s3measurements
+```
+
+## Inputs and outputs of the build pipeline
+
+The build pipeline takes as inputs:
+
+- a version number that is one of
+  - a release version number (e.g. `v2.2.0`) for release images
+  - a pseudo-version number (e.g. `v2.3.0-pre-debug`) for development images
+  - a pseudo-version number (e.g. `v2.3.0-pre-branch-name`) for branch images
+- a commit hash of the Constellation monorepo that is used to build the images (e.g. `cc0de5c68d41f31dd0b284d574f137e0b0ad106b`)
+
+To identify images belonging to one invocation of the build pipeline, the pipeline uses a unique identifier for the set of images, referred to as `image version uid` that combines the version number and the commit hash (e.g. `v2.2.0-cc0de5c68d41f31dd0b284d574f137e0b0ad106b`).
+
+The build pipeline produces as outputs:
+
+- a raw OS image for every CSP
+- a set of measurements for each raw OS image
+- one or more images uploaded to each CSP (e.g. AWS AMIs, Azure images, GCP images)
+- a lookup table that maps the `image version uid` to the CSP-specific image references
+
+The lookup table is uploaded to S3 and is used to identify the images that belong to a given `image version uid`.
+Measurements are uploaded to S3 and can be looked up for each cloud service provider and `image version uid`.
+
+## Image lookup table
+
+The image lookup table is a JSON file that maps the `image version uid` to the CSP-specific image references. It uses the `image version uid` as file name.
+
+```
+s3://<BUCKET-NAME>/images/<IMAGE-VERSION-UID>.json
+```
+
+```json
+{
+  "aws": {
+    "us-east-1": "ami-123",
+    "us-west-2": "ami-456",
+    "eu-west-1": "ami-789"
+  },
+  "azure": {
+    "cvm": "azure-cvm-123",
+    "trustedlaunch": "azure-trusted-launch-123"
+  },
+  "gcp": "gcp-image-123",
+  "qemu": "https://cdn.edgeless.systems/.../qemu.raw"
+}
+```
+
+- For AWS, the image lookup table contains the AMI IDs for each region
+- For Azure, the image lookup table contains the image IDs for the CVM and Trusted Launch images
+- For GCP, the image lookup table contains the image ID
+- For QEMU, the image lookup table contains a URL to the QEMU image
+
+This document is not signed and can be extended in the future to include more image references (e.g. if an image is replicated to a new AWS region).
+The same document can be used to identify old images that are no longer used and can be deleted for cost optimization.
+
+## Image measurements
+
+This RFC is not about the image measurements themselves, but about how they are stored and looked up.
+The format of the image measurements is described in the [secure software distribution RFC](secure-software-distribution.md).
+
+The image measurements are stored in a folder structure in S3 that is organized by CSP and `image version uid`.
+
+```
+s3://<BUCKET-NAME>/measurements/<CSP>/<IMAGE-VERSION-UID>/measurements.yaml
+s3://<BUCKET-NAME>/measurements/<CSP>/<IMAGE-VERSION-UID>/measurements.yaml.sig
```

Comment:
What do you think about having only one measurement file per version with all CSPs and images included that the CLI parses and compares against? 

The suggested format of the measurements would allow this with few modifications if we already keep CSP and image version in there.

This could be then simplified to `s3://<BUCKET-NAME>/measurements/<IMAGE-VERSION-UID>.yaml(.sig)`

Reply:
No major benefits, mainly just having one file instead of different paths. W

e wanted to revise the measurement pipeline anyway, so in case we also do Matrix uploads we could consolidate them. This would work similar to the release manifest, which in theory you could also split up in multiple files.